specs/flyout/flyout.md

```diff
@@ -0,0 +1,184 @@
+For more insight: [Managing focus in the shadow DOM](https://nolanlawson.com/2021/02/13/managing-focus-in-the-shadow-dom/)
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `hidden` - boolean, whether or not the flyout is hidden (defaults to false).
+- `trap-focus` - boolean, whether or not to keep focus contained inside the Flyout, defaults to true
+- `target` - string, the html id of the HTMLElement that the Flyout is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic, see *Position Options*
+- `delay` - number, milliseconds (ms) the time delay before the Flyout is shown once triggered
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+    ** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Properties*
+- `targetElement` - the reference to the HTMLElement used as target
+- `viewportElement` - the reference to the HTMLElement used as the viewport, required to set up anchored region properly
+- `flyoutVisible` - boolean, whether or not the Flyout is visible, for use inside the controller without changing exposed `hidden` attribute
+
+*Slots*
+- default for content
+
+*Events*
+- `dismiss` - event fired when the Flyout is dismissed
+
+*Functions*
+- `createFlyoutTemplate(string: prefix)` - generates a `ViewTemplate` for the Flyout based on the given provided prefix string. This is required as Flyout uses an `anchored-region` internally and the create function generates a template using the appropriate `anchored-region` tag (ie "fast-anchored-region", "fluent-anchored-region"). Note that the appropriate `anchored-region` component must also be declared.
+
+*Position Options*
```

Comment:
I'm not clear on the api for positioning.  What choices does the author have for this?   I'd also suggest that authors have the option to either force a position (locktodefault) or be based on available space.

Reply:
Anchored region can do "absolute" or "fixed" positioning.  I think we can elaborate on whether we want to present that choice in the api after we get an initial version in, but what do you see the default being?  "fixed" breaks out of parent containers more easily but absolute doesn't detach from the anchor as often.

Reply:
I updated the names, does this make more sense? These would be the options available.

I had a `responsive` attribute early on in this spec that would determine if the flyout was forced or would move based on available space, but removed it per some other comments.

I think having something like `responsive` (default to true) which means that the flyout would move along the same axis based upon available space is a great idea. If the flyout `position` option was "right" and `responsive` was true by default then when there was not enough space on the right of the target then the flyout would render on the left of the target. Does that make sense?

Reply:
I'm thinking we should simplify the number of positioning options available.   What if the choices were simply 'top', 'bottom', 'left' and right.  If an author chooses responsive their choice simply flips to the other side based on their choice (ie. left goes to right, top goes to bottom, etc...)?  My thinking is that our base components hava a simple positioning api, and if there is demand we can think about having a "pro" interface where advanced authors could pass a full "anchored region config" object and do all the things.